readthedocs/doc_builder/config.py

```diff
@@ -0,0 +1,114 @@
+
+from readthedocs_build.config import (ConfigError, BuildConfig,
+                                      load as load_config)
+
+
+class ConfigWrapper(object):
```

Comment:
This class has a lot of methods acting directly on the yaml returned dict. I thought this logic was to be implemented in `readthedocs-build`, is that not the case? This class could still exist as an abstract mapping class to the underlying shared config object produced by `readthedocs-build`, but it seems much of this would end up being implemented there anyways.